src/snapshot_tests/test_case.rs

```diff
@@ -216,6 +217,15 @@ impl TestCaseInstance {
     }
 }
 
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub enum TestInputPattern {
```

Comment:
I don't think we need to allow configuration here, but I think it's important to apply multiple effects at the same time